superset-frontend/src/explore/components/controls/MetricControl/AdhocMetricPopoverTrigger.tsx

```diff
@@ -33,7 +33,7 @@ export type AdhocMetricPopoverTriggerProps = {
   columns: { column_name: string; type: string }[];
   savedMetricsOptions: savedMetricType[];
   savedMetric: savedMetricType;
-  datasourceType: string;
+  datasource: Datasource;
```

Comment:
same comment about if this should be `datasource: Datasource` or `datasource?: Datasource`